File: backend/api/schemas/event.go
package schemas

import "github.com/graphql-go/graphql"

var eventType = graphql.NewObject(graphql.ObjectConfig{
	Name: "Event",
	Fields: graphql.Fields{
		"name": &graphql.Field{
			Type: graphql.String,
		},
		"description": &graphql.Field{
			Type: graphql.String,
		},
		"imageURL": &graphql.Field{
			Type: graphql.String,
		},
		"startTimestamp": &graphql.Field{
			Type: graphql.DateTime,
		},
		"endTimestamp": &graphql.Field{
			Type: graphql.DateTime,
		},
		"id": &graphql.Field{
			Type: graphql.ID,
		},
	},
})

func init() {
	eventType.AddFieldConfig("tickets", &graphql.Field{
		Type: graphql.NewList(ticketType),
	})
}

File: frontend/lib/backend/event/event.ts
type Event = {
    id: string,
    name: string,
    description: string,
    img_url: string,
    location: string,
    address: string,
    start_timestamp: Date,
    end_timestamp: Date
}

export function convertToEvent(rawData: { [key: string]: any }): Event {
    return {
        id: rawData.id,
        name: rawData.name,
        description: rawData.description,
        img_url: rawData.img_url,
        location: rawData.location,
        address: rawData.address,
        start_timestamp: new Date(rawData.start_timestamp),
        end_timestamp: new Date(rawData.end_timestamp)
    }
}

export default Event